spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+        {
+            val providerName: String = algAndProv[1];
+
+            action LOOP_FOR(
+                i, 0, providersListLength, +1,
+                findAlgorithmByAllProvidersByName_loop(i, providersList, _algorithm, resultProvider, resultAlgorithm, providerName)
+            );
+        }
+        else
+        {
+            action LOOP_FOR(
+                i, 0, providersListLength, +1,
+                findAlgorithmByAllProviders_loop(i, providersList, _algorithm, resultProvider, resultAlgorithm)
+            );
+        }
+
+        if (resultAlgorithm != null)
+        {
+            val isDefaultProvider: boolean = _isDefaultProvider(resultProvider);
+
+            result = new SecureRandomAutomaton(state = Initialized,
+                provider = resultProvider,
+                algorithm = resultAlgorithm,
+                defaultProvider = isDefaultProvider,
+            );
+
+            action LOOP_BREAK();
+        }
+    }
+
+
+    @static fun *.getSeed (numBytes: int): array<byte>
+    {
+        if (numBytes < 0)
+            _throwIAE();
+
+        _generateSeed(result, numBytes);
+    }
+
+
+    // methods
+
+    // within java.util.Random
+    fun *.doubles (@target self: SecureRandom): DoubleStream
+    {
+        val mass: array<double> = _generateRandomDoubleArray(MAX_RANDOM_STREAM_SIZE);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new DoubleStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.doubles (@target self: SecureRandom, randomNumberOrigin: double, randomNumberBound: double): DoubleStream
+    {
+        val mass: array<double> = _generateRandomDoubleArrayWithBounds(MAX_RANDOM_STREAM_SIZE, randomNumberOrigin, randomNumberBound);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new DoubleStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.doubles (@target self: SecureRandom, streamSize: long): DoubleStream
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<double> = _generateRandomDoubleArray(size);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new DoubleStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.doubles (@target self: SecureRandom, streamSize: long, randomNumberOrigin: double, randomNumberBound: double): DoubleStream
```

Comment:
What if `randomNumberBound` is **less** than `randomNumberOrigin`?

Reply:
Fixed;